### DISCUSSION TITLE
InnerSource Repository and Org Linter

### DISCUSSION THREAD

**Discussion by dellagustin-sap** [InnerSourceCommons/InnerSourcePatterns#730] [Pattern Ideas]
There is already some information available here: https://github.com/InnerSourceCommons/ispo-working-group/issues/79
There was also a presentation at the summit https://innersourcecommons.org/events/isc-2024/ about the SAP InnerSource linter.

**Comment by spier** [comment]
Copying the info from the Summit talk:

> Benjamin Ihrig (SAP SE) / Repository Linter@SAP
At SAP, InnerSource practices are increasingly becoming adopted as an essential part of our software development process, promoting collaboration, innovation and efficient code reuse. Ensuring consistency and quality across numerous repositories is challenging. To address this, we built an internal repository linter to standardize configurations and InnerSource-relevant content, while also providing automated fixes for detected issues. This session will share our journey of developing and piloting this tool to enhance repository quality and uniformity. Participants will gain insights into the rationale behind the linter, its implementation and the standards it aims to enforce.

**Comment by spier** [comment]
Is this idea more suitable for an entirely new pattern, or could it be expressed as multiple smaller extensions to existing patterns?

Elaborating on the latter:

For the [Standard Base Documentation](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/base-documentation.md) we could add text like

> You can make it easier for your users to stick to these standard files by offering a linter that will do XYZ

**Reply by dellagustin** [reply]
I have not thought too much about this yet, but I think it should be a dedicate pattern that links to other patterns.
